docs/reference/builder.md

```diff
@@ -1157,11 +1157,16 @@ For example, consider this Dockerfile:
 3 RUN echo $CONT_IMG_VER
 ```
 
-If you specify `--build-arg CONT_IMG_VER=<value>` on the command line the
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 RUN echo hello
+```
+If you specify `--build-arg CONT_IMG_VER=<value>` on the command line, in both cases, the
 specification on line 2 does not cause a cache miss; line 3 does cause a cache
 miss. The definition on line 2 has no impact on the resulting image. The `RUN`
 on line 3 executes a command and in doing so defines a set of environment
-variables, including `CONT_IMG_VER`. At that point, the `ARG` variable may
+variables, maybe including `CONT_IMG_VER`. At that point, the `ARG` variable may
```

Comment:
the "maybe" here could be confusing because the natural question would be "when is 'maybe' a yes and when is it no?".  Perhaps it would be better to say:
```
The `RUN` on line 3 executes a command and in doing so
defines a set of environment variables, including
`CONT_IMG_VER` if it has a value associated with it.
```